server/public/model/websocket_message.go

```diff
@@ -190,10 +208,38 @@ func (ev *WebSocketEvent) PrecomputeJSON() *WebSocketEvent {
 	return evCopy
 }
 
+func (ev *WebSocketEvent) RemovePrecomputedJSON() *WebSocketEvent {
+	evCopy := ev.Copy()
+	evCopy.precomputedJSON = nil
+	return evCopy
+}
+
+// RemoveHooks removes the broadcast hook information from the WebsocketBroadcast and returns it. It's intended
+// to be called before the WebSocketEvent is sent to the client.
+func (ev *WebSocketEvent) RemoveBroadcastHooks() (*WebSocketEvent, []string, []map[string]any) {
+	hooks := ev.broadcast.BroadcastHooks
+	hookArgs := ev.broadcast.BroadcastHookArgs
+
+	evCopy := ev.Copy()
+	evCopy.broadcast = ev.broadcast.copy()
+
+	evCopy.broadcast.BroadcastHooks = nil
+	evCopy.broadcast.BroadcastHookArgs = nil
+
+	return evCopy, hooks, hookArgs
+}
+
 func (ev *WebSocketEvent) Add(key string, value any) {
 	ev.data[key] = value
 }
 
+// AddWithCopy copies the map and writes to a copy of that,
+// and sets the map to the new event.
+func (ev *WebSocketEvent) AddWithCopy(key string, value any) {
+	ev.data = copyMap(ev.data)
```

Comment:
This looks potentially racy.

Reply:
At the moment, it's not because we copy the event before making any changes using it, but I could definitely see that being a risk in the future.

@agnivade This we should change this now? I could also add a comment to which of these methods aren't thread-safe as a warning

Reply:
I ended up removing this and just doing a deep copy of the event before the hooks process it. That seems safer, and the cost of deep copying is so minimal when a hook wants to make changes that I don't think it will matter